common/djangoapps/enrollment/views.py

```diff
@@ -421,7 +426,18 @@ def post(self, request):
                 )
 
             enrollment = api.get_enrollment(username, unicode(course_id))
-            if has_api_key_permissions and enrollment and enrollment['mode'] != mode:
+            mode_changed = enrollment and mode is not None and enrollment['mode'] != mode
+            active_changed = enrollment and is_active is not None and enrollment['is_active'] != is_active
+            if has_api_key_permissions and (mode_changed or active_changed):
+                if mode_changed and active_changed and not is_active:
+                    # if the requester wanted to deactivate but specified the wrong mode, fail
+                    # the request (on the assumption that the requester had outdated information
+                    # about the currently active enrollment).
+                    msg = u"Enrollment mode mismatch: active mode={0}, requested mode={1}. Won't deactivate.".format(
```

Comment:
Minor: You don't need the formatting placeholders in this format string. If you'd prefer to keep them, I'd use keywords instead of indices, for readability.